### PR TITLE
add lint config and update CI

### PR DESCRIPTION
**Pull request by hirosassa**
There are several "not-good" python codes on this repo such as `unused import`, `unused variables`, some formatting problems, etc.
To resolve above problem, I added Python linting CI and its configurations.

### CODE REVIEW

File: pyrightconfig.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "reportOptionalSubscript": false,
+  "reportOptionalMemberAccess": false
+}
```
Comment (hirosassa) [inline pyrightconfig.json:4]: This configuration is for reducing useless `pyright` errors

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ignore = E741,W503
+extend-ignore = E203
+exclude = .git,.github,__pycache__,rust,shell,data
+max-line-length = 160
+
+# too high. need to reduce complexity
+max-complexity = 33
+
+[isort]
+profile = black
```
Comment (hirosassa) [inline tox.ini:52]: `tox.ini` is the overall configuration for testing and linting.

File: README.md [hunk 3]
```diff
@@ -202,3 +202,9 @@ at your option.
 Unless you explicitly state otherwise, any contribution intentionally submitted
 for inclusion in the work by you, as defined in the Apache-2.0 license, shall be
 dual licensed as above, without any additional terms or conditions.
+
+Before you submit your PR, please keep in mind the following practices:
+
+1. Use `tox -e fmt` to format the code.
+1. Use `tox -e lint` to lint and typecheck
+1. Use `tox` to run all tests and lint
```
Comment (hirosassa) [inline README.md:210]: This is how to use on your local PC.

**Review by hirosassa — COMMENTED** [review]
@kg86 Please review my changes.

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    pipenv run black --check src
+
+[flake8]
+ignore = E741,W503
+extend-ignore = E203
+exclude = .git,.github,__pycache__,rust,shell,data
+max-line-length = 160
+
+# too high. need to reduce complexity
+max-complexity = 33
```
Comment (hirosassa) [inline tox.ini:49]: This values is the maximum allowed McCabe complexity value for a block of code.
ref:
https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-max-complexity

Currently I set `33` but it should be 10 or lower.
We need to improve the code.

File: README.md (hunk 3 above)
Comment (kg86) [reply README.md:210]: Nice instruction!

**Review by kg86 — APPROVED** [review]

**Comment by kg86** [comment]
Thank you for your contribution. This must improve the development productivity.

**Pull request merged**